sensor/tag_based_pnp_calibrator/README.md

```diff
@@ -0,0 +1,116 @@
+# tag_based_pnp_calibrator
+
+A tutorial for this calibrator can be found [here](../docs/tutorials/tag_based_pnp_calibrator.md)
+
+## Purpose
+
+The package `tag_based_pnp_calibrator` allows extrinsic calibration among the camera sensor and lidar sensor used in autonomous driving and robotics.
+
+## Inner-workings / Algorithms
+
+The `tag_based_pnp_calibrator` utilizes the PnP algorithm to calculate the transformation between the camera and lidar. To run this package, you also need to operate the `apriltag_ros` package and the `lidartag` package to calculate the transformation.
+
+The `apriltag_ros` package detects the AprilTag and outputs the detection results. Conversely, the `lidartag` package detects the LidarTag and outputs its detection results.
+
+The `tag_based_pnp_calibrator` utilizes the detections from both `apriltag_ros` package and `lidartag` package, employing a Kalman Filter to track these detections. If the detections converge, the calibrator applies the SQPnP algorithm provided by OpenCV to estimate the transformation between the image points from AprilTag and the object points from LidarTag.
+
+### Diagram
+
+Below, you can see how the algorithm is implemented in the `tag_based_pnp_calibrator` package.
+
+![segment](../docs/images/tag_based_pnp_calibrator/tag_based_pnp_calibrator.jpg)
```

Comment:
This diagram is not clear at all in what are the relationship between elements. For example, it is not an UML, so you need to explain the kind of connections you have (I get the feeling that different kind of connections are used with the same line type). I will only write it here, but probably applies to all your diagrams